### PR TITLE
Improve site packages directory retrieval across platforms #11949

### LINKED ISSUE #11949: Frontend Config Settings not correctly set on Windows causing Arches App assets to not be found
In `settings_utils.py`, the `generate_frontend_configuration` function uses  `site.getsitepackages()[0]` to fetch the site-packages directory path that is used to load dependencies.

On linux/mac, `site.getsitepackages()` returns a list, with one entry like...
```py
 [
   "/usr/local/lib/python3.12/site-packages"
]
```
but on Windows it returns two entries...
```py
[
   "E:\\arches_system\\env",
   "E:\\arches_system\\env\\Lib\\site-packages"
]
```
Therefore, on Windows the `SITE_PACKAGES_DIRECTORY` value is not correctly set. This seems to work fine when there is no Arches app added to the Arches project as there seem to be a lot of try catches that handle the project.

So for Windows, it would seem that we either select the second list item, or it is hardened to check for the first item with `site-packages` in the path. I am unsure however, if this list may vary depending on the environment.

### PR DESCRIPTION
**Pull request by aj-he**
### Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

Hardens the site-packages directory path lookup for Windows users. Arches app frontend assets cannot be found on Windows otherwise.

### Issues Solved

Closes #11949

### Checklist

-   I targeted one of these branches:
    - [ ] dev/8.0.x (under development): features, bugfixes not covered below
    - [x] dev/7.6.x (main support): regressions, crashing bugs, security issues, major bugs in new features
    - [ ] dev/6.2.x (extended support): major security issues, data loss issues
-   [x] I added a changelog in arches/releases
-   [ ] I submitted a PR to arches-docs (if appropriate)
-   [x] Unit tests pass locally with my changes
-   [x] I added tests that prove my fix is effective or that my feature works
-   [ ] My test fails on the target branch

#### Accessibility Checklist

[Developer Guide](https://arches.readthedocs.io/en/stable/developing/advanced/accessibility/)

| Topic            | Changed | Retested |
| ---------------- | ------- | -------- |
| Color contrast   |         |          |
| Form fields      |         |          |
| Headings         |         |          |
| Links            |         |          |
| Keyboard         |         |          |
| Responsive Design|         |          |
| HTML validation  |         |          |
| Screen reader    |         |          |

#### Ticket Background
*   Sponsored by: Historic England
*   Found by: @aj-he 
*   Tested by: @aj-he 
*   Designed by: @aj-he

### CODE REVIEW

File: arches/settings_utils.py [hunk 1]
```diff
@@ -152,12 +152,22 @@ def generate_frontend_configuration():
...
+        if not site_packages_dir and site_packages_paths:
+            site_packages_dir = site_packages_paths[0]
+
         frontend_configuration_settings_data = {
             "_comment": "This is a generated file. Do not edit directly.",
             "APP_ROOT": app_root_path,
             "ARCHES_APPLICATIONS": arches_app_names,
             "ARCHES_APPLICATIONS_PATHS": path_lookup,
-            "SITE_PACKAGES_DIRECTORY": site.getsitepackages()[0],
+            "SITE_PACKAGES_DIRECTORY": site_packages_dir,
```
Comment (jacobtylerwalls) [inline arches/settings_utils.py]: Thanks for finding this bug. I wonder if we can just do:
```suggestion
            "SITE_PACKAGES_DIRECTORY": sysconfig.get_path("purelib"),
```

Comment (aj-he) [reply arches/settings_utils.py]: @jacobtylerwalls - I'll take a look at this to see what happens on Windows/Linux etc.

Comment (aj-he) [reply arches/settings_utils.py]: @jacobtylerwalls - yep... worked a charm and a lot more concise.

**Comment by aj-he** [comment]
@jacobtylerwalls - you ok to review this for me? I've added a small test for the fix as it was failing test coverage. I did see that the function has been moved in 8.x but hopefully not too tricky to merge?

**Comment by jacobtylerwalls** [comment]
Certainly, I'll take a look likely tomorrow. Only to suggest some simplifications to the tests.

File: releases/7.6.10.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ### Bug Fixes and Enhancements
 
 - Fix regression in original fix (#11748) for slow reports when used on systems with many related resources [#11959](https://github.com/archesproject/arches/pull/11959)
+- Fixes frontent build issue on Windows [#11949](https://github.com/archesproject/arches/pull/11949)
```
Comment (jacobtylerwalls) [inline releases/7.6.10.md]:
```suggestion
- Fixes frontend build issue on Windows [#11949](https://github.com/archesproject/arches/pull/11949)
```

File: releases/7.6.10.md [hunk 3]
```diff
@@ -28,7 +29,24 @@ JavaScript:
     pip install --upgrade arches==7.6.10
     ```
 
-3. If you are running Arches on Apache, restart your server:
+3.  If you are upgrading from version 7.6.9, you'll need to rebuild your frontend bundle:
```
Comment (jacobtylerwalls) [inline releases/7.6.10.md:33]: Is this strictly true? The frontend changes are addressing edge cases that caused bundles to not build at all, so if you've got a bundle, wouldn't you be all set?

File: tests/utils/test_settings_utils.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    ):
+        """
+        Test frontend configuration generation when running core arches directly.
+        """
+        # Mock settings
+        mock_settings = mock.MagicMock()
+        mock_settings.APP_ROOT = "/arches/app"
+        mock_settings.ROOT_DIR = "/arches"
+        mock_settings.PUBLIC_SERVER_ADDRESS = "http://localhost:8000"
+        mock_settings.STATIC_URL = "/static/"
```
Comment (jacobtylerwalls) [inline tests/utils/test_settings_utils.py]: I'd like to stick to the existing pattern for overriding settings during tests via the `@override_settings` decorator and thereby remove as much mocking as possible.
```suggestion
```

File: tests/utils/test_settings_utils.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+        mock_settings.WEBPACK_DEVELOPMENT_SERVER_PORT = 8080  # Add this missing setting
```
Comment (jacobtylerwalls) [inline tests/utils/test_settings_utils.py]: This setting isn't missing from the test settings.

File: tests/utils/test_settings_utils.py [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+from django.test import TestCase, override_settings
+from django.conf import settings
+from arches.settings_utils import generate_frontend_configuration
+
+
+# these tests can be run from the command line via
+# python manage.py test tests.utils.test_settings_utils --settings="tests.test_settings"
+
+
+class TestFrontendConfigurationGeneration(TestCase):
```
Comment (jacobtylerwalls) [inline tests/utils/test_settings_utils.py:16]: Thanks for adding this. Now that we have a test in place, let's keep ourselves accountable by no longer excluding it from the coverage calculation. Would you mind adjusting the `settings*.py` to `*settings.py` or similar?

File: tests/utils/test_settings_utils.py [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+            {"@/arches/*", "@/test_app/*", "*"},
+        )
+
+    @mock.patch("arches.settings_utils.sys.stderr.write")
+    def test_generate_frontend_configuration_error_handling(self, mock_stderr_write):
+        """
+        Test error handling in frontend configuration generation.
+        """
+        # Force an exception by removing a required setting
+        with mock.patch("arches.settings_utils.settings") as mock_settings:
```
Comment (jacobtylerwalls) [inline tests/utils/test_settings_utils.py]: Same as above, you can use `@override_settings` or a variant like `self.modify_settings()`.

File: tests/utils/test_settings_utils.py [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+            delattr(mock_settings, "APP_ROOT")
+
+            # This should raise an AttributeError
+            with self.assertRaises(Exception):
```
Comment (jacobtylerwalls) [inline tests/utils/test_settings_utils.py]:
```suggestion
            with self.assertRaises(AttributeError):
```

File: tests/utils/test_settings_utils.py [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+        args, _ = mock_json_dump.call_args_list[1]
+        self.assertEqual(
+            set(args[0]["_comment"]), set(expected_tsconfig_data["_comment"])
+        )
+        self.assertEqual(
+            set(args[0]["compilerOptions"]["paths"].keys()),
+            {"@/arches/*", "@/test_app/*", "*"},
+        )
+
+    @mock.patch("arches.settings_utils.sys.stderr.write")
```
Comment (jacobtylerwalls) [inline tests/utils/test_settings_utils.py]: We can avoid mocking with the more common:
```py
with captured_stderr() as stderr:
   # write to stderr
   self.assertIn('foo', stderr.getvalue())
```

File: tests/utils/test_settings_utils.py [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+            generate_frontend_configuration()
+
+        # Check file paths
+        mock_open.assert_any_call(
+            "/arches/../.frontend-configuration-settings.json", "w"
+        )
+        mock_open.assert_any_call("/arches/../.tsconfig-paths.json", "w")
+
+        # Check content of first file
+        expected_settings_data = {
```
Comment (jacobtylerwalls) [inline tests/utils/test_settings_utils.py:34]: I wouldn't be afraid of removing much of the mocking and writing this file to a tempdir, but I leave that to you.

**Review by jacobtylerwalls — CHANGES_REQUESTED** [review]
Thanks for the updates, just a few suggestions in the tests, let me know what you think. Mostly I just want to avoid having the tests look forbidding to the next developer :-)

File: releases/7.6.10.md (hunk 3 above)
Comment (jacobtylerwalls) [reply releases/7.6.10.md:33]: Ah, I take it the arches apps were omitted, perhaps silently? Then let's leave this in. 👍

File: releases/7.6.10.md (hunk 2 above)
Comment (aj-he) [reply releases/7.6.10.md]: done

File: tests/utils/test_settings_utils.py (hunk 7 above)
Comment (aj-he) [reply tests/utils/test_settings_utils.py]: done

File: tests/utils/test_settings_utils.py (hunk 10 above)
Comment (aj-he) [reply tests/utils/test_settings_utils.py:34]: I've stripped most out now

File: tests/utils/test_settings_utils.py (hunk 9 above)
Comment (aj-he) [reply tests/utils/test_settings_utils.py]: done

File: tests/utils/test_settings_utils.py (hunk 6 above)
Comment (aj-he) [reply tests/utils/test_settings_utils.py:16]: @jacobtylerwalls - I'm slightly confused by this. When I submitted my initial change, the coverage check failed the PR despite me only changing the `settings_utils.py` and `release/7.6.x.md` files. The above note suggests that `settings_utils.py` would have been excluded? This is the only reason I added the test!!

File: tests/utils/test_settings_utils.py (hunk 5 above)
Comment (aj-he) [reply tests/utils/test_settings_utils.py]: done

File: tests/utils/test_settings_utils.py (hunk 4 above)
Comment (aj-he) [reply tests/utils/test_settings_utils.py]: applied

File: tests/utils/test_settings_utils.py (hunk 6 above)
Comment (aj-he) [reply tests/utils/test_settings_utils.py:16]: ... also, if I include `settings_utils.py` in the coverage at this point, am I going to have to write tests for the other functions in the module for this to be accepted?

Comment (jacobtylerwalls) [reply tests/utils/test_settings_utils.py:16]: Oh I certainly wouldn't ask you to volunteer for testing a bunch of unrelated stuff, no worries.

>  the coverage check failed the PR despite me only changing the settings_utils.py

I can't speak to why that comparison failed, but I wonder if there was a slight drift between your branch and upstream dev/7.6.x

If you go to the actions tab and look at the coverage reports generated, you'll see there's no arches/settings_utils.py being measured

Comment (jacobtylerwalls) [reply tests/utils/test_settings_utils.py:16]: The coverage check has to be taken with a grain of salt because of the "falling denominator" problem -- it's always acceptable to remove tested code.

**Review by jacobtylerwalls — APPROVED** [review]

File: tests/utils/test_settings_utils.py (hunk 6 above)
Comment (aj-he) [reply tests/utils/test_settings_utils.py:16]: thanks for reviewing and sorting out @jacobtylerwalls.
